### PR TITLE
Fix issue 23151 (Cleanup TODO introduced by PR #23148)

### PR DESCRIPTION
**Pull request by cshung**

### CODE REVIEW

**Comment by cshung** [comment]
Test Ubuntu x64 Checked Innerloop Build and Test (Jit - TieredCompilation=0)

**Comment by hoyosjs** [comment]
@dotnet-bot test Ubuntu x64 Checked Innerloop Build and Test
@dotnet-bot test Ubuntu x64 Checked Innerloop Build and Test (Jit - TieredCompilation=0)

File: src/vm/eventpipebuffer.cpp [hunk 1]
```diff
@@ -75,6 +75,15 @@ bool EventPipeBuffer::WriteEvent(Thread *pThread, EventPipeSession &session, Eve
 
         // Placement-new the EventPipeEventInstance.
         // if pthread is NULL, it's likely we are running in something like a GC thread which is not a Thread object, so it can't have an activity ID set anyway
+
+        StackContents s;
```
Comment (jorive) [inline src/vm/eventpipebuffer.cpp]: I believe you can initialize memory to zero like this: `StackContents s{};`

**Comment by cshung** [comment]
@dotnet-bot test Ubuntu x64 Checked Innerloop Build and Test
@dotnet-bot test Ubuntu x64 Checked Innerloop Build and Test (Jit - TieredCompilation=0)

**Comment by cshung** [comment]
@dotnet-bot Test Ubuntu x64 Checked Innerloop Build and Test (Jit - TieredCompilation=0)

**Comment by cshung** [comment]
@dotnet-bot test Ubuntu x64 Checked Innerloop Build and Test
@dotnet-bot test Ubuntu x64 Checked Innerloop Build and Test (Jit - TieredCompilation=0)

**Comment by cshung** [comment]
@dotnet-bot test Ubuntu x64 Checked Innerloop Build and Test

**Comment by MichalStrehovsky** [comment]
Nit: would it be possible to follow the [Contributor guidelines](https://github.com/dotnet/coreclr/blob/030a3ea9b8dbeae89c90d34441d4d9a1cf4a7de6/Documentation/project-docs/contributing.md#commit-messages) with respect to commit messages? "Fix issue 23040", "Fix issue 23037", "Fix issue 23009" make for [very poor history](https://github.com/dotnet/coreclr/commits?author=cshung).

Also, please make sure to squash your commits, especially if they're like in #23143.

**Comment by cshung** [comment]
@dotnet-bot test Ubuntu x64 Checked Innerloop Build and Test

**Pull request merged**